src/main/java/org/apache/sysml/api/python/SystemML.py

```diff
@@ -259,25 +260,31 @@ def getStringRDD(self, varName, format):
         #    traceback.print_exc()
 
 def getNumCols(numPyArr):
-	if numPyArr.ndim == 1:
-		return 1
-	else:
-		return numPyArr.shape[1]
+    if numPyArr.ndim == 1:
+        return 1
+    else:
+        return numPyArr.shape[1]
        
-def convertToJavaMatrix(sc, src):
-	src = np.asarray(src)
-	numCols = getNumCols(src)
-	numRows = src.shape[0]
-	arr = src.ravel().astype(np.float64)
-	buf = bytearray(arr.tostring())
-	return sc._jvm.org.apache.sysml.runtime.instructions.spark.utils.RDDConverterUtilsExt.convertPy4JArrayToMB(buf, numRows, numCols)
-	
+def convertToMatrixBlock(sc, src):
+    if isinstance(sc, SparkContext):
+        src = np.asarray(src)
+        numCols = getNumCols(src)
+        numRows = src.shape[0]
+        arr = src.ravel().astype(np.float64)
+        buf = bytearray(arr.tostring())
+        return sc._jvm.org.apache.sysml.runtime.instructions.spark.utils.RDDConverterUtilsExt.convertPy4JArrayToMB(buf, numRows, numCols)
+    else:
+        raise TypeError('sc needs to be of type SparkContext') # TODO: We can generalize this by creating py4j gateway ourselves
```

Comment:
I say we stay tied to Spark for now as the end user doesn't gain anything using SystemML in a standalone mode, as opposed to running it in Spark local mode on a laptop.